### PR TITLE
zig-wrapper: compile with explicit `-target`

### PR DESCRIPTION
**Pull request by motiejus-wix**
Sometimes the on-host toolchain may not be detected correctly on OSX[1]. Since we are interested only in a *hermetic* toolchain, we are meant to use cross-compilation mode by invoking `-target`. Let's do it.

[1]: https://github.com/ziglang/zig/issues/13313#issuecomment-1761496151

### CODE REVIEW

**Comment by motiejus-wix** [comment]
cc @daliuss @pauliusjwix

**Review by pauliusjwix — APPROVED** [review]

File: toolchain/defs.bzl [hunk 1]
```diff
@@ -174,11 +175,13 @@ def _zig_repository_impl(repository_ctx):
...
         "build-exe",
+        "-target",
+        _TARGET_MCPU[host_platform][0],
+        "-mcpu={}".format(_TARGET_MCPU[host_platform][1]),
         "-fstrip",
-        "-mcpu={}".format(_MCPU[host_platform]),
         "-OReleaseSafe",
         "zig-wrapper.zig",
-    ] + (["-static"] if os == "linux" else [])
+    ]
```
Comment (laurynaslubys) [inline toolchain/defs.bzl:184]: I assume that `-static` for linux got dropped on purpose?

Comment (motiejus) [reply toolchain/defs.bzl:184]: Yes, musl impliea that.

**Review by motiejus — APPROVED** [review]
Approving my own, maybe this will help github understand it's reviewed?

**Pull request merged**